src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1426,46 +1433,56 @@ public Map<String, List<Pool>> mapPoolsBySubscriptionIds(Collection<String> subs
         return output;
     }
 
-    @SuppressWarnings("unchecked")
-    public CandlepinQuery<Pool> getPoolsBySubscriptionId(String subId) {
+    public List<Pool> getPoolsBySubscriptionId(String subId) {
```

Comment:
This method can call into `getPoolsBySubscriptionIds` with a list of the single element provided so we only have one query to maintain, both literally and figuratively.

Reply:
Good catch! Yeah, that's way better. Updated